system/Cache/Handlers/FileHandler.php

```diff
@@ -229,24 +229,32 @@ public function isSupported(): bool
      * Does the heavy lifting of actually retrieving the file and
      * verifying it's age.
      *
-     * @return array|bool|float|int|object|string|null
+     * @return array<string, mixed>|false
+     * @phpstan-return array{data: mixed, ttl: int, time: int}|false
      */
     protected function getItem(string $filename)
     {
         if (! is_file($this->path . $filename)) {
             return false;
         }
 
+        /** @phpstan-var array{data: mixed, ttl?: int, time?: int}|mixed~array $data */
```

Comment:
What is `mixed~array`?

Reply:
the `mixed` type excluding `array`. this notation is used by phpstan. i think this come from the bitwise `~`

Reply:
Removed the inline var for now. It seems on higher rule levels this gives phpdoc-parser errors.